05_Datathon/diabetes_prediction_example.Rmd

```diff
@@ -0,0 +1,336 @@
+- [TableOne documentation](https://cran.r-project.org/web/packages/tableone/tableone.pdf)
+
+```{r tableone, echo=TRUE}
+library(tableone)
+
+allvars = c("bmi", "age", "gender", "ethnicity","glucose_apache","icu_type")
+catvars = c("ethnicity","gender", "icu_type")
+
+table_gossis <- CreateTableOne(vars = allvars, data = train_set, 
+                factorVars = catvars,
+                strata="target_variable")
+
+kableone(table_gossis)
+```
+
+## 5. Model Construction
+
+### 5.1 Train a random forest classifier
+
+- We will use the `randomForest` library to build a random forest model.
+- See the [randomForest documentation on CRAN](https://cran.r-project.org/web/packages/randomForest/randomForest.pdf).
+
+```{r train_model, echo=TRUE}
+# Define forest tuning parameters
+
+# 5-fold cross validation
+control <- trainControl(method='repeatedcv', 
+                        number=5, 
+                        repeats=3)
+
+# Number of variables tried at each split
+mtry <- sqrt(ncol(train_set))
+
+# Grid search is a linear search through a vector of mtry values
+tunegrid <- expand.grid(.mtry=mtry)
+
+# Create classification forest using age, bmi, and gender
+forest <- randomForest(target_variable ~ age + bmi + gender + glucose_apache, 
+              data = train_set,
+              metric = "Accuracy",
+              tuneGrid = tunegrid,
+              trControl = control)
+```
+
+### 5.2 View the variable importance
+
+- Many approaches for computing "importance".
+- `rpart` calculates a metric using the [sum of contributions to splits]((https://cran.r-project.org/web/packages/rpart/vignettes/longintro.pdf)).
+- Shapley Additive exPlanations (ShAP) values are a popular alternative.
+- Compute the marginal contribution of a feature across a combination of features.
+
+```{r importance, echo=TRUE}
+library(knitr)
+kable(forest$importance)
+```
+We can see that `glucose_apache` and `bmi` had the most predictive power.
+
+
+### 5.3 Predict diabetes mellitus in five unseen patients from test set
+
+#### Select unseen patients from testing set
+
+- Let's look five unseen patients in our test set.
+- What outcome do you expect for these patients?
+
+```{r select_unseen, echo=TRUE}
+unseen <- head(test_set, 5) %>% 
+          select(age, bmi, gender, glucose_apache)
+
+print(unseen)
+```
+
+#### Predict the diabetes mellitus in unseen patients
+
+- Is the model correct for this patient?
+- `type="prob"` to output probabilities
+- `type="class"` to output classes
+
+```{r predict_unseen, echo=TRUE}
+# Make predictions on testing set
+forest_pred <- predict(forest,
+                     newdata = test_set,
+                     type="class")
+# Combine unseen patients data with corresponding predictions
+data.frame(age = unseen$age,bmi = unseen$bmi,gender = unseen$gender, glucose_apache = unseen$glucose_apache, prediction = forest_pred[1:5], truth_value = test_set$target_variable[1:5]) %>% kable()
```

Comment:
you may need to select the second column of the caret predictions, as it computes both class predictions (I believe)